src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php

```diff
@@ -406,16 +415,35 @@ public function getWriteInfo(string $class, string $property, array $context = [
             return null;
         }
 
+        $allowAdderRemover = $context['enable_adder_remover_extraction'] ?? true;
+        $propertyMetadata = $this->getClassMetadata($class)?->getPropertyMetadataFor($property);
+        $adderAccessName = $propertyMetadata?->adder;
+        $removerAccessName = $propertyMetadata?->remover;
+
+        if ($allowAdderRemover && null !== $adderAccessName && null !== $removerAccessName) {
+            $adderMethod = new \ReflectionMethod($class, $adderAccessName);
+            $removerMethod = new \ReflectionMethod($class, $removerAccessName);
+
+            $mutator = new PropertyWriteInfo(PropertyWriteInfo::TYPE_ADDER_AND_REMOVER);
+            $mutator->setAdderInfo(new PropertyWriteInfo(PropertyWriteInfo::TYPE_METHOD, $adderAccessName, $this->getWriteVisibilityForMethod($adderMethod), $adderMethod->isStatic()));
+            $mutator->setRemoverInfo(new PropertyWriteInfo(PropertyWriteInfo::TYPE_METHOD, $removerAccessName, $this->getWriteVisibilityForMethod($removerMethod), $removerMethod->isStatic()));
+
+            return $mutator;
+        }
+
+        if (null !== $methodName = $propertyMetadata?->setter) {
+            $method = new \ReflectionMethod($class, $methodName);
+
+            return new PropertyWriteInfo(PropertyWriteInfo::TYPE_METHOD, $methodName, $this->getWriteVisibilityForMethod($method), $method->isStatic());
+        }
+
         $allowGetterSetter = $context['enable_getter_setter_extraction'] ?? false;
         $magicMethods = $context['enable_magic_methods_extraction'] ?? $this->magicMethodsFlags;
         $allowMagicCall = (bool) ($magicMethods & self::ALLOW_MAGIC_CALL);
         $allowMagicSet = (bool) ($magicMethods & self::ALLOW_MAGIC_SET);
         $allowConstruct = $context['enable_constructor_extraction'] ?? $this->enableConstructorExtraction;
-        $allowAdderRemover = $context['enable_adder_remover_extraction'] ?? true;
 
-        $camelized = $this->camelize($property);
```

Comment:
No point in getting the camelized version if the info can be extracted from the constructor.